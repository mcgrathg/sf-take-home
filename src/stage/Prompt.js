import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import colors from '../config/colors';

const StyledPrompt = styled.h1`
  margin: 0 auto 1rem;
  color: ${colors.greyDark};
  font-size: 1.5rem;
  line-height: 1.75;
  font-weight: normal;
`;

const Highlighted = styled.span`
  position: relative;
  font-weight: 800;

  &::after {
    position: absolute;
    left: 0;
    bottom: 0;
    height: 0.5rem;
    width: 100%;
    content: '';
    background-color: ${colors.teal};
    z-index: -1;
    transition: all 0.5s;
    transform-origin: left top;
    transform: scale(0, 1);
  }

  &.animate::after {
    transform: scale(1, 1);
  }
`;

class Prompt extends Component {
  state = {
    animate: false,
  };

  isMounted = false;

  componentDidMount() {
    this.isMounted = true;

    setTimeout(() => {
      if (this.isMounted) {
        this.setState({ animate: true });
      }
    }, 500);
  }

  componentWillUnmount() {
    this.isMounted = false;
  }

  render() {
    const { prompt, highlight, promptEnd } = this.props;
    const { animate } = this.state;
    return (
      <StyledPrompt>
        {prompt}
        <Highlighted className={`${animate ? 'animate' : ''}`}>
          {highlight}
        </Highlighted>
        {promptEnd}
      </StyledPrompt>
    );
  }
}

Prompt.propTypes = {
  prompt: PropTypes.string,
  highlight: PropTypes.string,
  promptEnd: PropTypes.string,
};

Prompt.defaultProps = {
  prompt: '',
  highlight: '',
  promptEnd: '?',
};

export default Prompt;
