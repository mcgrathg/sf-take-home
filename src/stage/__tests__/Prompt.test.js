import React from 'react';
import { render } from 'react-testing-library';
import Prompt from '../Prompt';

describe('Prompt', () => {
  it('renders', () => {
    const { container } = render(<Prompt />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
