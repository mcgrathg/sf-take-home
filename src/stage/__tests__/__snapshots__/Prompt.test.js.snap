// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prompt renders 1`] = `
.c0 {
  margin: 0 auto 1rem;
  color: #2f3237;
  font-size: 1.5rem;
  line-height: 1.75;
  font-weight: normal;
}

.c1 {
  position: relative;
  font-weight: 800;
}

.c1::after {
  position: absolute;
  left: 0;
  bottom: 0;
  height: 0.5rem;
  width: 100%;
  content: '';
  background-color: #ccf2f0;
  z-index: -1;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
  -webkit-transform-origin: left top;
  -ms-transform-origin: left top;
  transform-origin: left top;
  -webkit-transform: scale(0,1);
  -ms-transform: scale(0,1);
  transform: scale(0,1);
}

.c1.mounted::after {
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

<h1
  class="c0"
>
  
  <span
    class="c1"
  />
  ?
</h1>
`;
